pytest-pylint no longer works with pytest >= 3.7.0

Pytest deprecated storing items on the pytest module, so we need to replace it with a new pattern

For others looking for this issue, here's the error message:

```
$ pytest --pylint --pylint-rcfile=.pylintrc -m pylint core/
============================= test session starts ==============================
platform linux -- Python 3.6.6, pytest-3.7.0, py-1.5.4, pluggy-0.7.1
rootdir: /builds/sms/core, inifile:
plugins: pylint-0.12.0
collected 2 items / 1 errors
-----------------------------------------------------------------
Linting files

-----------------------------------------------------------------

==================================== ERRORS ====================================
________________ ERROR collecting core/videoreader/__init__.py _________________
/opt/conda/envs/exec-core/lib/python3.6/site-packages/pluggy/hooks.py:258: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
/opt/conda/envs/exec-core/lib/python3.6/site-packages/pluggy/manager.py:67: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
/opt/conda/envs/exec-core/lib/python3.6/site-packages/pluggy/manager.py:61: in <lambda>
    firstresult=hook.spec_opts.get('firstresult'),
/opt/conda/envs/exec-core/lib/python3.6/site-packages/pytest_pylint.py:146: in pytest_collect_file
    if parent.pylint_config is None:
E   AttributeError: 'Package' object has no attribute 'pylint_config'
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
=========================== 1 error in 0.77 seconds ============================
```

Current workaround, just rollback `pytest==3.6.4` until moving to `pytest==3.7.0` is done.

--- a/pytest_pylint.py
+++ b/pytest_pylint.py
@@ -137,21 +137,22 @@ def pytest_sessionstart(session):
 
 def pytest_collect_file(path, parent):
     """Collect files on which pylint should run"""
-    config = parent.config
+    config = parent.session.config
     if not config.option.pylint or config.option.no_pylint:
         return None
     if path.ext != ".py":
         return None
-    rel_path = get_rel_path(path.strpath, parent.fspath.strpath)
-    if parent.pylint_config is None:
-        parent.pylint_files.add(rel_path)
+    rel_path = get_rel_path(path.strpath, parent.session.fspath.strpath)
+    session = parent.session
+    if session.pylint_config is None:
+        session.pylint_files.add(rel_path)
         # No pylintrc, therefore no ignores, so return the item.
         return PyLintItem(path, parent)
 
-    if not any(basename in rel_path for basename in parent.pylint_ignore):
-        parent.pylint_files.add(rel_path)
+    if not any(basename in rel_path for basename in session.pylint_ignore):
+        session.pylint_files.add(rel_path)
         return PyLintItem(
-            path, parent, parent.pylint_msg_template, parent.pylintrc_file
+            path, parent, session.pylint_msg_template, session.pylintrc_file
         )
     return None
 

